src/layoutBuilder.js

```diff
@@ -388,75 +495,288 @@ function convertWordsToCodepoints(lineAsArrayOfCodepoints) {
+	// level.
+
+	// Contains the codepoints for each word in the line.
+	var arrayOfCodePoints = convertWordsToCodepoints(bidiString.string_arr);
+
+	// Contains the strings for each word in the line.
+	var arrayOfTransformedWords = [];
+
+	// With the line converted to an array of words, with the words represented as
+	// codepoints, the words must be individually re-run through the BIDI algorithm
+	// to reverse the characters in each RTL (as the initial BIDI process reverses each
+	// character in the RTL words).
+
+	// Pass the codepoints for each word through String.fromCharCode
+	for (
+		var groupIndex = 0;
+		groupIndex < arrayOfCodePoints.length;
+		groupIndex++
+	) {
+		// Get current word as a string
+		var groupString = spreadify(
+			String.fromCharCode,
+			String
+		)(arrayOfCodePoints[groupIndex]);
+		// Run the word through BIDI to reorder the characters
+		var bidiWord = bidi.from_string(groupString, { direction: 'RTL' });
+		bidiWord.reorder_visually();
+
+		// Push the word to what will be the final line array
+		arrayOfTransformedWords.push(bidiWord.toString());
+	}
+
+	// Pass the transformed words, as a single string, through the buildInlines utility.
+	// Include original styling of the node as a whole e.g. if the whole body of text is bold
+	// then make sure this gets passed through.
+	var updatedInlines = textTools.buildInlines(
+		[{ text: arrayOfTransformedWords.join(''), style: textNode.style }],
+		styleStack
+	);
+
+	// Wipe the existing words (inlines) from this line ...
+	line.inlines = [];
+	// ... and replace them with our new BIDI-ified, reversed inlines
+	updatedInlines.items.forEach(function (inline) {
+		// Where the BIDI algorithm has appropriately transformed the content
+		// we can be confident that the postions of the words have been reversed.
+		// For example, a word that was at position 0 prior to the transformation will
+		// now likely be at the end of the array. This gives us a way of extracting the
+		// styling for each word before it was transformed and apply it to the
+		// transformed word.
+
+		// TODO I don't know how this will stack up again LTR words. Will need
+		// to evaluate once we have mixed fonts support.
+
+		// This needs to use find because the arrays inlinesBeforeTransformation and 
+		// updatedInlines are not necessarily the same lengths (due to how the reversal
+		// interacts with whitespace and punctuation in some cases)
+		var oldInline = inlinesBeforeTransformation.find((oldInline) =>
+			inline.text.includes(oldInline.text.trim())
+		);
+		var newInline = inline;
+		if (oldInline) {
+			newInline.style = oldInline.style;
+			newInline.background = oldInline.background;
+			newInline.font = oldInline.font;
+			newInline.decoration = oldInline.decoration;
+			newInline.decorationColor = oldInline.decorationColor;
+		}
+		return line.addInline(newInline);
+	});
+}
+
+function convertWordsToInlineCodepoints(lineAsArrayOfCodepoints) {
```

Comment:
Would some of these functions benefit from unit tests?

Reply:
Yes